www/class/centreon-clapi/centreonConfiguration.class.php

```diff
@@ -0,0 +1,333 @@
+        foreach ($stmt->fetchAll(\PDO::FETCH_ASSOC) as $value) {
+            if ($value['service_register'] === '0') {
+                $serviceTemplateIds[] = $value['service_service_id'];
+            } elseif ($value['hostgroup_hg_id'] !== null) {
+                $hostgroupIds[] = $value['hostgroup_hg_id'];
+            } else {
+                $hostIds[] = $value['host_host_id'];
+            }
+        }
+
+        $serviceIds = $this->getServicesForConfigChangeFlagFromServiceTemplateIds($serviceTemplateIds);
+
+        return array_merge(
+            $hostIds,
+            $this->getHostsForConfigChangeFlagFromHostgroupIds($hostgroupIds),
+            $this->getHostsForConfigChangeFlagFromServiceIds($serviceIds)
+        );
+    }
+
+    /**
+     * Return ids of pollers linked to hosts
+     *
+     * @param int[] $hostIds
+     * @param bool $shouldHostBeEnabled (default true)
+     * @return int[]
+     * @throws \Exception
+     */
+    public function getPollersForConfigChangeFlagFromHostIds(array $hostIds, bool $shouldHostBeEnabled = true): array
+    {
+        if (empty($hostIds)) {
+            return [];
+        }
+
+        $bindedParams = [];
+        foreach ($hostIds as $key => $hostId) {
+            $bindedParams[':host_id_' . $key] = $hostId;
+        }
+
+        if ($shouldHostBeEnabled) {
+            $query = "SELECT DISTINCT(phr.nagios_server_id)
+            FROM ns_host_relation phr
+            JOIN host ON host.host_id = phr.host_host_id
+            WHERE host.host_activate = '1' AND phr.host_host_id IN (" . implode(', ', array_keys($bindedParams)) . ")";
+        } else {
+            $query = "SELECT DISTINCT(phr.nagios_server_id) FROM ns_host_relation phr
+            WHERE phr.host_host_id IN (" . implode(', ', array_keys($bindedParams)) . ")";
+        }
+
+        $stmt = $this->db->prepare($query);
+        foreach ($bindedParams as $bindedParam => $bindedValue) {
+            $stmt->bindValue($bindedParam, $bindedValue, \PDO::PARAM_INT);
+        }
+        $stmt->execute();
+
+        return $stmt->fetchAll(\PDO::FETCH_COLUMN);
+    }
+
+    /**
+     * Set 'updated' flag to '1' for all listed poller ids
+     *
+     * @param int[] $pollerIds
+     * @throws \Exception
+     */
+    private function setPollersToUpdated(array $pollerIds): void
+    {
+        if (empty($pollerIds)) {
+            return;
+        }
+
+        $bindedParams = [];
+        foreach ($pollerIds as $key => $pollerId) {
+            $bindedParams[':poller_id_' . $key] = $pollerId;
+        }
+        $query = "UPDATE nagios_server SET updated = '1' WHERE id IN ("
+            . implode(', ', array_keys($bindedParams)) . ")";
+        $stmt = $this->db->prepare($query);
+        foreach ($bindedParams as $bindedParam => $bindedValue) {
+            $stmt->bindValue($bindedParam, $bindedValue, \PDO::PARAM_INT);
+        }
+        $stmt->execute();
+    }
+
+    /**
+     * Undocumented function
```

Comment:
```suggestion
     * Undocumented function
```
document this function